tests/cephfs/cephfs_mirroring/test_cephfs_mirroring_configure_cephfs_mirroring.py

```diff
@@ -0,0 +1,333 @@
+import random
+import string
+import time
+import traceback
+
+from tests.cephfs.cephfs_mirroring.cephfs_mirroring_utils import CephfsMirroringUtils
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    CEPH-83574099 - Validate the Synchronisation is successful upon enabling fs mirroring.
+    This function performs a series of operations to test CephFS mirroring.
+    It has pre-requisites,
+    performs various operations, and cleans up the system after the tests.
+
+    Args:
+        ceph_cluster: The Ceph cluster to perform the mirroring tests on.
+        **kw: Additional keyword arguments.
+
+    Returns:
+        int: 0 if the test is successful, 1 if there's an error.
+
+    Raises:
+        Exception: Any unexpected exceptions that might occur during the test.
+    """
+
+    try:
+        config = kw.get("config")
+        ceph_cluster_dict = kw.get("ceph_cluster_dict")
+        fs_util_ceph1 = FsUtils(ceph_cluster_dict.get("ceph1"))
+        fs_util_ceph2 = FsUtils(ceph_cluster_dict.get("ceph2"))
+        fs_mirroring_utils = CephfsMirroringUtils(
+            ceph_cluster_dict.get("ceph1"), ceph_cluster_dict.get("ceph2")
+        )
+        build = config.get("build", config.get("rhbuild"))
+        source_clients = ceph_cluster_dict.get("ceph1").get_ceph_objects("client")
+        target_clients = ceph_cluster_dict.get("ceph2").get_ceph_objects("client")
+        cephfs_mirror_node = ceph_cluster_dict.get("ceph1").get_ceph_objects(
+            "cephfs-mirror"
+        )
+
+        log.info("checking Pre-requisites")
+        if not source_clients or not target_clients:
+            log.info(
+                "This test requires a minimum of 1 client node on both ceph1 and ceph2."
+            )
+            return 1
+        fs_util_ceph1.prepare_clients(source_clients, build)
+        fs_util_ceph2.prepare_clients(target_clients, build)
+        fs_util_ceph1.auth_list(source_clients)
+        fs_util_ceph2.auth_list(target_clients)
+        source_fs = "cephfs"
+        target_fs = "cephfs"
+        target_user = "mirror_remote"
+        target_site_name = "remote_site"
+
+        log.info("Enable mirroring mgr module on source and destination")
+        fs_mirroring_utils.enable_mirroring_module(source_clients[0])
```

Comment:
In enable_mirroring_module we are just logging it as an error if there is any failure.
In the test case also we are not validating and proceeding to further steps. 
Do we need to fail at this point if the enabling of mirroring fails and stop executing it further?

Holds good for other functions as well

Reply:
@AmarnatReddy  : I am planning to use this condition in the test case, let me know if I am validating correctly. 
``
result = fs_mirroring_utils.enable_mirroring_module(source_clients[0])
if result:
    log.info("Mirroring module enabled successfully.")
else:
    log.error("Failed to enable mirroring module. Test aborted.")
    return 1
``